### ISSUE TITLE
chatglm2-6B使用script/chat/single_chat.py下的脚本推理失败，显示输出乱码。

### ISSUE THREAD

**Issue by NingJinzhong** [yangjianxin1/Firefly#90]
![image](https://github.com/yangjianxin1/Firefly/assets/38681542/cf0bee85-f9e4-4cb9-9c58-77b8580b16d0)

**Comment by FeiWard** [comment]
是的，我的也是乱码

**Comment by FeiWard** [comment]
这个我重新摸索了一下，解决方案主要有两个，放出来大家讨论
1、增加数据量，我是拿了另外一个任务的数据，量会比较大，10W+，经过微调，batchsize=4，训练1500步，结果正常
2、增加迭代次数，小数据集500条+，增加迭代轮次，loss降低
{'loss': 9.7175, 'learning_rate': 9.833333333333334e-07, 'epoch': 3.31}                                      
{'loss': 4.3647, 'learning_rate': 1.9833333333333335e-06, 'epoch': 6.63}
看了下配置文件，因为warmup的默认设置是3000，我这个数据量根本无法达到，所以这段时间反而破坏了模型自有的能力，所以增加迭代轮次能够让模型学习到数据本身的东西

其他还在摸索中，有新的发现再同步出来

**Comment by NingJinzhong** [comment]
> 是的，我的也是乱码
chatglm2推理的时候用官方提供的response, history = model.chat()方式就没乱码了。

**Comment by timxx** [comment]
> 这个我重新摸索了一下，解决方案主要有两个，放出来大家讨论 1、增加数据量，我是拿了另外一个任务的数据，量会比较大，10W+，经过微调，batchsize=4，训练1500步，结果正常 2、增加迭代次数，小数据集500条+，增加迭代轮次，loss降低 {'loss': 9.7175, 'learning_rate': 9.833333333333334e-07, 'epoch': 3.31} {'loss': 4.3647, 'learning_rate': 1.9833333333333335e-06, 'epoch': 6.63} 看了下配置文件，因为warmup的默认设置是3000，我这个数据量根本无法达到，所以这段时间反而破坏了模型自有的能力，所以增加迭代轮次能够让模型学习到数据本身的东西
> 
> 其他还在摸索中，有新的发现再同步出来

你是把迭代次数加到超过这个warmup_steps吗？
